samples/Luau/none.luau

```diff
@@ -0,0 +1,21 @@
+--// Authored by @benbrimeyer (https://github.com/benbrimeyer)
+--// Fetched from (https://github.com/duckarmor/Freeze/blob/670bb6593d8cc54cdcbb96cd94d1273ee0420abb/source/None.luau)
+--// Licensed under the MIT License (https://github.com/duckarmor/Freeze/blob/670bb6593d8cc54cdcbb96cd94d1273ee0420abb/LICENSE)
+
+--[=[
+	@prop None None
+	@within Freeze
+
+	Since lua tables cannot distinguish between values not being present and a value of nil,
+	`Freeze.None` exists to represent values that should be interpreted as `nil`.
+
+	This is useful when removing values with functions such as [`Freeze.Dictionary.merge`](../api/Dictionary#merge).
+]=]
```

Comment:
@lildude: based on your comment on #6616, should the tokenizer be taught about `--[[ ]]` and `--[====[`/`]====]` comments? Or is that only necessary if there are extension ambiguities?

Reply:
There's probably no harm in adding support for this format but it only really comes into play when there are extension ambiguities.